Helper/ProductSerializer.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Metrilo\Analytics\Helper;
+
+class ProductSerializer extends \Magento\Framework\App\Helper\AbstractHelper
+{
+    /**
+     * Product parent types
+     */
+    const PARENT_TYPES = [\Magento\ConfigurableProduct\Model\Product\Type\Configurable::TYPE_CODE, \Magento\Bundle\Model\Product\Type::TYPE_CODE, \Magento\GroupedProduct\Model\Product\Type\Grouped::TYPE_CODE];
+    
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface                   $storeManager,
+        \Magento\Bundle\Model\Product\Type                           $bundleType,
+        \Magento\GroupedProduct\Model\Product\Type\Grouped           $groupedType,
+        \Magento\ConfigurableProduct\Model\Product\Type\Configurable $configurableType,
+        \Metrilo\Analytics\Model\ProductData                         $productData
+    ) {
+        $this->storeManager     = $storeManager;
+        $this->bundleType       = $bundleType;
+        $this->groupedType      = $groupedType;
+        $this->configurableType = $configurableType;
+        $this->productData      = $productData;
+    }
+    
+    public function serialize($product)
+    {
+        $productId     = $product->getId();
+        $productType   = $product->getTypeId();
+        $storeId       = $product->getStoreId();
+    
+        if ($product->getData('observer_update')) {
```

Comment:
I don't think it is any of the Serializer's business to make these decisions for parent/child products, can this be cleared from the serializer? Tell me what do you think

Reply:
I can move all this parent/child logic to additional productRelation helper for example and just return the productOptions data, if any.

Reply:
This is still unsolved

Reply:
All product options logic is now moved to separate helper classes for product options and product image respectively.